lib/protractor.js

```diff
@@ -661,34 +661,30 @@ Protractor.prototype.initDebugger_ = function(debuggerClientPath, opt_debugPort)
     return asString;
   };
 
-  if (opt_debugPort) {
-    process.debugPort = opt_debugPort;
-  }
-
-  // Call this private function instead of sending SIGUSR1 because Windows.
-  process._debugProcess(process.pid);
-
-  var flow = webdriver.promise.controlFlow();
-  var self = this;
-  var pausePromise = flow.execute(function() {
-    log.puts('Starting WebDriver debugger in a child process. Pause is ' +
-        'still beta, please report issues at github.com/angular/protractor\n');
-    var args = [process.debugPort];
-    if (self.debuggerServerPort_) {
-      args.push(self.debuggerServerPort_);
-    }
-    var nodedebug = require('child_process').fork(debuggerClientPath, args);
-    process.on('exit', function() {
-      nodedebug.kill('SIGTERM');
+  // Invoke fn if port is available.
+  var onPortAvailable = function(port, fn) {
+    var net = require('net');
+    var tester = net.connect({port: port}, function() {
+      console.error('Port ' + port + ' is already in use. Please specify ' + 
+        'another port to debug.');
+      process.exit(1);
     });
-  });
+    tester.once('error', function (err) {
+      if (err.code === 'ECONNREFUSED') {
+        tester.once('close', fn).end();
+      } else {
+        console.error('Unexpected failure testing for port ' + port + ': ', 
+          err);
+        process.exit(1);
+      }
+    });
+  };
 
   var vm_ = require('vm');
+  var flow = webdriver.promise.controlFlow();
+
   var context = { require: require };
-  for (var key in global) {
-    context[key] = global[key];
-  }
-  context.list = function(locator) {
+  global.list = function(locator) {
```

Comment:
I see. I don't think we need autocomplete to suggest it, since it's all of 4 characters, but I don't really care either way.